### PR TITLE
Add scripts for detecting DDL atomicity issues

### PR DESCRIPTION
**Pull request by fizaaluthra**

### CODE REVIEW

File: ddl_atomicity/ddl_atomicity_check_script_python2.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        pg_oid = pg_class_entry['oid']
+        
+        if tablename != pg_class_entry['relname']:
+            print("❌ - Table {} with oid {} and uuid {} exists in {} but has a mismatched table name: table name in YSQL metadata is {} - TABLE NAME NEEDS TO BE FIXED".format(tablename, pg_oid, tableid, dbname, pg_class_entry['relname']))
+            errorcount += 1
+        else:
+            print("✅ - Table {} with oid {} and uuid {} exists in database {} - NO ISSUE".format(tablename, pg_oid, tableid, dbname))
+
+        # Get columns
+        table_schema_json = json.loads(subprocess.check_output(["curl", "-s", "http://{}:7000/api/v1/table?id={}".format(webserver_interface, tableid)]).decode())
```
Comment (ionthegeek) [inline ddl_atomicity/ddl_atomicity_check_script_python2.py:94]: PATH

**Review by ionthegeek — COMMENTED** [review]
There are some portions of this script that will fail on customer systems because they have changed install paths, port numbers, etc.

File: ddl_atomicity/ddl_atomicity_check_script_python2.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+    # Add table to the database's list in the dictionary
+    if dbname not in db_tables:
+        db_tables[dbname] = []
+    db_tables[dbname].append((table["table_name"], pg_oid, yb_pg_table_oid, table["uuid"]))
+
+# Iterate through each database
+for dbname, tables in db_tables.items():
+    # Fetch all user tables from pg_class for the database
+    pg_class_output = json.loads(subprocess.check_output([ysqlsh_path, "-h", "/tmp/.yb.0.0.0.0:5433/", "-d", dbname, "-t", "-c", "SELECT json_agg(row_to_json(t)) FROM (SELECT relname, oid, relfilenode FROM pg_class WHERE oid >= 16384) t;"]).decode().strip())
```
[the remark this answers is not in the document]
Comment (fizaaluthra) [reply ddl_atomicity/ddl_atomicity_check_script_python2.py:57]: Yeah, I was told we can't use python libraries that aren't pre-installed.

File: ddl_atomicity/ddl_atomicity_check_script_python2.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import sys
+import subprocess
+
+# Initialize error count
+errorcount = 0
+
+FNULL = open(os.devnull, 'w')
+html_parser = HTMLParser.HTMLParser()
+
+ysqlsh_path = "/home/yugabyte/tserver/bin/ysqlsh"
```
[the remark this answers is not in the document]
Comment (fizaaluthra) [reply ddl_atomicity/ddl_atomicity_check_script_python2.py]: Right, I used these hard-coded paths as I wasn't sure how this script is going to be run and had only my local environment readily available for testing. I have parameterized all of this now, thanks!

File: ddl_atomicity/ddl_atomicity_check_script_python2.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+is_leader = subprocess.check_output(r"curl -s http://localhost:9300/metrics | grep yb_node_is_master_leader\{ | awk '{print $2}'", shell=True).decode('utf-8').strip()
```
[the remark this answers is not in the document]
Comment (fizaaluthra) [reply ddl_atomicity/ddl_atomicity_check_script_python2.py]: I'm not sure, this was in place for YBM clusters -- I have changed this now to only run if --master_leader_only is set.

File: ddl_atomicity/ddl_atomicity_check_script_python2.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+error_messages = []
+
+FNULL = open(os.devnull, 'w')
+html_parser = HTMLParser.HTMLParser()
+
+# Check if the current node is the master leader if the flag is set
+if args.master_leader_only:
+    print("Checking if the node is the master leader.")
+    is_leader = subprocess.check_output(
+        r"curl -s http://localhost:9300/metrics | grep yb_node_is_master_leader\{ | awk '{print $2}'",
```
Comment (ionthegeek) [inline ddl_atomicity/ddl_atomicity_check_script_python2.py:32]: The paths to `curl`, `grep`, `awk`, etc. should be set as variables or called using `command -p <binary>` (assuming Python's `subprocess` module is shelling out to `bash`, anyway) to ensure the script doesn't break if PATH is unset or incorrect.

File: ddl_atomicity/ddl_atomicity_check_script_python2.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    with open(args.master_conf_path, 'r') as f:
+        master_conf = f.read()
+        master_interface_address = master_conf.split('webserver_interface=')[1].split('\n')[0]
+else:
+    master_interface_address = args.master_interface_address
+
+# Get table data
+tables_output = json.loads(
+    subprocess.check_output(
+        ["curl", "-s", "http://{}:{}/api/v1/tables".format(master_interface_address, args.master_interface_port)]
```
Comment (ionthegeek) [inline ddl_atomicity/ddl_atomicity_check_script_python2.py:51]: PATH

File: ddl_atomicity/ddl_atomicity_check_script_python2.py [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+            print(
+                "✅ - Table {} with oid {} and uuid {} exists in database {} - NO ISSUE".format(
+                    tablename, pg_oid, tableid, dbname
+                )
+            )
+
+        # Get columns
+        table_schema_json = json.loads(
+            subprocess.check_output([
+                "curl",
```
Comment (ionthegeek) [inline ddl_atomicity/ddl_atomicity_check_script_python2.py:147]: PATH

File: ddl_atomicity/ddl_atomicity_check_script_python2.py [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+            errorcount += 1
+        else:
+            print(
+                "✅ - Table {} with oid {} and uuid {} exists in database {} - NO ISSUE".format(
+                    tablename, pg_oid, tableid, dbname
+                )
+            )
+
+        # Get columns
+        table_schema_json = json.loads(
```
Comment (ionthegeek) [inline ddl_atomicity/ddl_atomicity_check_script_python2.py:145]: What happens if the output from the `curl` comes back as something other than JSON?

**Review by ionthegeek — CHANGES_REQUESTED** [review]
My previous comment about PATH and calling binaries by name remains unaddressed.

File: ddl_atomicity/ddl_atomicity_check_script_python2.py (hunk 5 above)
Comment (fizaaluthra) [reply ddl_atomicity/ddl_atomicity_check_script_python2.py:32]: Ah I misunderstood your previous comment, thanks for clarifying, will update

**Review by ionthegeek — APPROVED** [review]

**Pull request merged**